### ISSUE TITLE
Awesome! ffmpeg_all_codec.js

### ISSUE THREAD

**Issue by Noitidart** [inbasic/media-converter#16]
Hi there!
This is really cool. I didn't realize ffmpeg was in use by another firefox addon. I made an addon last week which records the screen. WebRTC records as `.webm` but stuff like Twitter only accepts gif or mp4. Things are working great EXCEPT when I have audio in the webm recording. If there is audio, then converting it to mp4 fails. I was wondering if you could please take a look at these topics - https://github.com/bgrins/videoconverter.js/issues/49

Here is my addon repository - https://github.com/Noitidart/Screencastify/

Here is my released addon: https://github.com/mozilla/addons/issues/145#issue-161592099

I couldn't solve this conversion to mp4, if you could help that would be awesome. I set up a demo here:
http://noitforks.github.io/videoconverter.js/demo/

You can run any command on this page. But if you click "Video to mp4" you will see the error I am getting.

**Comment by inbasic** [comment]
@Noitidart the FFmpeg used in `videoconverter.js` is outdated. For `webm` conversions you need a newer version of FFmpeg. Try the executable and it works just fine; https://github.com/inbasic/media-converter/blob/master/src/lib/config.js#L3

P.S: `videoconverter.js` is not performing very well on Firefox. Why don't you use the FFmpeg executable instead?

**Comment by Noitidart** [comment]
Aw drats, thanks for that. Is it possible to pack with my addon a cross platform executable? I know exe works for Windows, but what about for mac and linux?

**Comment by inbasic** [comment]
I don't recommend packing ffmpeg. You can ask users to install FFmpeg or even you can offer downloading it like I did in this extension.

or alternatively, you can push a pull request to Media Converter to support MP4 conversion and then use this extension in background for your conversion requests; https://github.com/inbasic/media-converter/blob/master/connect.md

> but what about for mac and linux

See the JSON response of the feed URL. FFmpeg executable is available on all these three platforms.

**Comment by Noitidart** [comment]
Thanks so much! I try to make my addons have 0 dependencies. May you please point me to the cross platform executables?
Seriously appreciate this!

**Comment by Noitidart** [comment]
Holy heck I think I found them, they are like 40mb each and for linux they dont have one solution. Shoot. I think Ill try to push ffmpeg.js guys to update :(

**Comment by inbasic** [comment]
yes, there is not a single cross platform executable. You need to offer platform dependent executables; see browser_download_url's of https://api.github.com/repos/inbasic/ffmpeg/releases/latest

> think Ill try to push ffmpeg.js guys to update

Have you tried to convert an actual video file using the ffmpeg.js? Firefox literally cannot handle it (Chrome is much better though).

**Comment by Noitidart** [comment]
Yep it works first class. I do it off main thread from a ChromeWorker. Only
issue is converting webm with audio to mp4. It even updates a message box
on all progress as it converts its pretty sweet!

**Comment by Noitidart** [comment]
Screencast using my addon to show you ffmpeg in firefox woohoo :d https://www.youtube.com/watch?v=gCTFPmDmHwE

**Comment by inbasic** [comment]
Nice! Thanks for sharing

**Comment by Noitidart** [comment]
@inbasic by  chance do you know of any alternative js libraries to convert webm files?

**Comment by inbasic** [comment]
Not that I am aware of. You can also try to compile the webm part of ffmpeg using https://github.com/kripken/emscripten

**Comment by Noitidart** [comment]
Oh interesting, thanks very much!
